ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java

```diff
@@ -116,43 +165,124 @@ private void collectData(File[] replicas, ArrayList<String> replicasList, Map<St
-                  + "\n" + consistencyBlobResult);
+          potentialInconsistentBlobs.add(blobId);
+          earliestRealInconsistentBlobTimeMs =
+              Math.min(earliestRealInconsistentBlobTimeMs, consistencyBlobResult.getOpTime());
         }
       }
     }
+    logger.info("The latest op executed at {} ms.", latestOpTimeMs);
+    logInconsistentBlobs(potentialInconsistentBlobs, blobIdToStatusMap, replicaCount, realInconsistentBlobs,
+        inconsistentDueToReplicationCount);
+    // Inconsistent blobs = real inconsistent + acceptable inconsistent + inconsistent due to replication Lag
+    // Acceptable inconsistent = due to deletion, some replicas reports as deleted, whereas some reports as unavailable
+    // Inconsistent due to replication lag = Inconsistency due to replication lag.
+    // Anything else is considered to be real inconsistent blobs
     logger.info("Total Inconsistent blobs count : " + inconsistentBlobs);
     if (includeAcceptableInconsistentBlobs) {
       logger.info("Acceptable Inconsistent blobs count : " + acceptableInconsistentBlobs);
     }
-    logger.info("Real Inconsistent blobs count :" + realInconsistentBlobs);
-    return realInconsistentBlobs == 0;
+    logger.info(
+        "Inconsistent due to replication lag : {}. Difference between first inconsistent and lastestOpTime is {} ms",
+        inconsistentDueToReplicationCount.get(), latestOpTimeMs - earliestRealInconsistentBlobTimeMs);
+    logger.info("Real Inconsistent blobs count :" + realInconsistentBlobs.size());
+    if (realInconsistentBlobs.size() > 0) {
+      logger.info("The earliest real inconsistent blob had its last operation at {} ms",
+          earliestRealInconsistentBlobTimeMs);
+    }
+  }
+
+  /**
+   * Figure out real inconsistent blobs from those due to replication lag and print
+   * @param potentialInConsistentblobs {@link List} of potential inconsistent blobs
+   * @param blobIdToStatusMap {@link Map} of blobId to {@link BlobStatus}
+   * @param replicaCount total replica count
+   * @param realInconsistentBlobs {@link List} of real inconsistent blobs
+   * @param inconsistentDueToReplicationCount {@link AtomicLong} to keep track of inconsistent blobs count due to
+   *                                                            replication lag
+   */
+  private void logInconsistentBlobs(List<String> potentialInConsistentblobs, Map<String, BlobStatus> blobIdToStatusMap,
+      int replicaCount, List<String> realInconsistentBlobs, AtomicLong inconsistentDueToReplicationCount) {
+    for (String potentialInConsistentBlob : potentialInConsistentblobs) {
+      BlobStatus consistencyBlobResult = blobIdToStatusMap.get(potentialInConsistentBlob);
+      if (consistencyBlobResult.belongsToRecentIndexSegment()) {
+        if ((consistencyBlobResult.getAvailableList().size() + consistencyBlobResult.getUnavailableList().size()
```

Comment:
Why is that? Consider the following
1. Server 1 receives a put for blob B.
2. Server 2 replicates blob B.
3. Server 1 receives a delete for blob B.
4. Server 3 has not replicated either the put or the delete from Server 1 (either because it has not got there or it has seen the put + delete so it ignored the blob).

In this case server 1 would have the blob in deleted state, server 2 in put state and server 3 in unavailable state. Is this set of events not possible?